### ISSUE TITLE
Tinyproxy not running after reboot

### ISSUE THREAD

**Issue by nopdotcom** [StreisandEffect/streisand#1383]
Reproducible #1381 

### Expected behavior:

`systemctl show tinyproxy` shows running tinyproxy processes

```
$ systemctl status tinyproxy
● tinyproxy.service - tinyproxy - a light-weight HTTP/HTTPS proxy daemon for POS
   Loaded: loaded (/etc/systemd/system/tinyproxy.service; enabled; vendor preset
   Active: active (running) since Mon 2018-06-18 00:33:13 UTC; 55s ago
     Docs: man:tinyproxy(8)
           https://www.banu.com/tinyproxy/
  Process: 17566 ExecStop=/usr/bin/killall -9 tinyproxy (code=exited, status=0/S
  Process: 17572 ExecStart=/usr/sbin/tinyproxy -c /etc/tinyproxy.conf (code=exit
 Main PID: 17577 (tinyproxy)
    Tasks: 11
   Memory: 3.2M
      CPU: 14ms
   CGroup: /system.slice/tinyproxy.service
           ├─17577 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17578 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17579 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17580 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17581 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17582 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17583 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17584 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17585 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           ├─17586 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
           └─17587 /usr/sbin/tinyproxy -c /etc/tinyproxy.conf
```

### Actual Behavior:

After reboot:

```
# systemctl status tinyproxy
● tinyproxy.service - tinyproxy - a light-weight HTTP/HTTPS proxy daemon for POSIX operating systems
   Loaded: loaded (/etc/systemd/system/tinyproxy.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: resources) since Mon 2018-06-18 00:35:22 UTC; 62ms ago
     Docs: man:tinyproxy(8)
           https://www.banu.com/tinyproxy/
  Process: 2051 ExecStart=/usr/sbin/tinyproxy -c /etc/tinyproxy.conf (code=exited, status=0/SUCCESS)

Jun 18 00:35:22 ip-172-31-25-200 systemd[1]: tinyproxy.service: Daemon never wrote its PID file. Failing.
Jun 18 00:35:22 ip-172-31-25-200 systemd[1]: Failed to start tinyproxy - a light-weight HTTP/HTTPS proxy daemon for POSIX operating systems.
Jun 18 00:35:22 ip-172-31-25-200 systemd[1]: tinyproxy.service: Unit entered failed state.
Jun 18 00:35:22 ip-172-31-25-200 systemd[1]: tinyproxy.service: Failed with result 'resources'.
ubuntu@ip-172-31-25-200:~
```

Five minutes later it remains in this state.

Something really is keeping it from writing its PID to the pidfile.

#1381 

### Steps to Reproduce:
1. Install
1. Reboot

### Ansible Information

* Ansible version: 2.5.4
* Ansible system: Linux
* Host OS: Ubuntu
* Host OS version:  16.04
* Python interpreter: python
* Python version: 2.7.12

### Streisand Information

* Streisand Git revision: e5ba596fef3299b3a0887aca14f1d558163f2144
* Streisand Git clone has untracked changes: no
* Genesis role: genesis-amazon
* Custom SSH key: False

### Enabled Roles

* Shadowsocks enabled:  True
* Wireguard enabled: True
* OpenVPN enabled: False
* stunnel enabled: False
* Tor enabled: False
* Openconnect enabled: False
* TinyProxy enabled: True
* SSH forward user enabled: True
* Configured number of VPN clients: 5

**Comment by tiliarou** [comment]
I have the same issue.
What are the permissions of the tinyproxy process/service ?

**Comment by tiliarou** [comment]
I tried to investigate this on tiny proxy side and found:
[https://github.com/tinyproxy/tinyproxy/issues/106](https://github.com/tinyproxy/tinyproxy/issues/106)
[https://github.com/tinyproxy/tinyproxy/pull/120](https://github.com/tinyproxy/tinyproxy/pull/120)

If I understand correctly, we don't have to use this PID file anymore ? So we could just configure streisand not to use tinyproxy PID option ?

**Comment by tiliarou** [comment]
From what I understood, many guides advise to change PID path to tmp since tinyproxy drops privileges prior to generating the PID file (for security reason beyond my understanding).

One more thing, which version are we installing with apt ? latest release on github is from 2016, whereas there was commits from 22 days ago...

**Comment by alimakki** [comment]
@tiliarou tinyproxy is being installed from Ubuntu's universe repo, see [package details](https://packages.ubuntu.com/xenial/tinyproxy)

**Comment by tiliarou** [comment]
It's 1.8.3.3, which seem outdated considérant 1.8.4  was released in 2016...
Maybe compiling this by Streisand would solve issues ?
Or it's just that tinyproxy was not meant to be used with var path

**Comment by alimakki** [comment]
Resolved as of #1391
